common4j/src/main/com/microsoft/identity/common/java/net/HttpRequest.java

```diff
@@ -0,0 +1,101 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.net;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.NonNull;
+
+/**
+ * Internal class representing a http request.
+ */
+public class HttpRequest {
+
+    private static final String HOST = "Host";
+
+    private final URL mRequestUrl;
+
+    private final byte[] mRequestContent;
+
+    URL getRequestUrl() {
+        return mRequestUrl;
+    }
+
+    byte[] getRequestContent() {
+        return mRequestContent;
+    }
+
+    String getRequestContentType() {
+        return mRequestContentType;
+    }
+
+    String getRequestMethod() {
+        return mRequestMethod;
+    }
+
+    Map<String, String> getRequestHeaders() {
+        return Collections.unmodifiableMap(mRequestHeaders);
+    }
+
+    private final String mRequestContentType;
+    private final String mRequestMethod;
+    private final Map<String, String> mRequestHeaders = new HashMap<>();
+
+    /**
+     * Constructor for {@link HttpRequest} with request {@link URL}, headers, post message and the
+     * request content type.
+     *
+     * @param requestUrl         The {@link URL} to make the http request.
+     * @param requestHeaders     Headers used to send the http request.
+     * @param requestContent     Post message sent in the post request.
+     * @param requestContentType Request content type.
+     */
+    public HttpRequest(@NonNull final URL requestUrl,
+                @NonNull final Map<String, String> requestHeaders,
+                @NonNull final String requestMethod,
+                final byte[] requestContent,
+                final String requestContentType) {
+        mRequestUrl = requestUrl;
+        mRequestHeaders.put(HOST, requestUrl.getAuthority());
+        mRequestHeaders.putAll(requestHeaders);
+        mRequestMethod = requestMethod;
+        mRequestContent = requestContent;
+        mRequestContentType = requestContentType;
+    }
+
+    /**
+     * Check if the given status code is the retryable status code(500/503/504).
+     *
+     * @param statusCode The status to check.
+     * @return True if the status code is 500, 503 or 504, false otherwise.
+     */
+    public static boolean isRetryableError(final int statusCode) {
```

Comment:
>isRetryableError [](http://example.com/codeflow?start=26&length=16)

This doesnt seem to belong HttpRequest.
Maybe HttpResponse or a separate Utility class if we have one.

Reply:
I'm going to put this in UrlConnectionHttpClient.
Once we remove the adapter classes, that will be the only place this function is going to be called.